bgp/modules/terms.py

```diff
@@ -113,3 +115,50 @@ def validate_isbn(isbn):
     def __init__(self):
         super(IsbnExtractorModule, self).__init__(self.validate_isbn)
 
+
+class PageTypeProcessor:
+
+    def __init__(self, modules):
+        self.modules = modules
+    
+    def run(self, book):
```

Comment:
This run method should essentially be a double-for loop.

The outer-for loop will iterate over each page (like you're doing within the pageProcessor method).

The inner for loop will send each xml page to the each module and capture the results :)

Reply:
Think of the PageTypeProcessor as something which takes a book, splits the book into pages, and then makes sure each of its modules gets to perform its function on the page. PageTypeProcess will then accumulate the results from each module so that they may be accessed later.